addons/account/models/account_journal_dashboard.py

```diff
@@ -700,67 +685,37 @@ def _get_draft_sales_purchases_query(self):
-            ('move_id.payment_state', 'in', ('not_paid', 'partial')),
-            ('account_type', '=', 'asset_receivable'),
-            ('move_id.move_type', '=', 'out_invoice'),
-            ('parent_state', '=', 'posted'),
-        ])
-
-    def _get_late_sales_query(self):
+    def _get_open_sale_purchase_query(self, journal_type):
+        assert journal_type in ('sale', 'purchase')
         return self.env['account.move.line']._where_calc([
-            *self.env['account.move.line']._check_company_domain(self.env.companies),
-            ('journal_id', 'in', self.ids),
-            ('move_id.payment_state', 'in', ('not_paid', 'partial')),
-            ('account_type', '=', 'asset_receivable'),
-            ('date_maturity', '<', fields.Date.context_today(self)),
-            ('move_id.move_type', '=', 'out_invoice'),
-            ('parent_state', '=', 'posted'),
-        ])
-
-    def _get_open_purchases_query(self):
-        return self.env['account.move.line']._where_calc([
-            *self.env['account.move.line']._check_company_domain(self.env.companies),
-            ('journal_id', 'in', self.ids),
-            ('move_id.payment_state', 'in', ('not_paid', 'partial')),
-            ('account_type', '=', 'liability_payable'),
-            ('move_id.move_type', '=', 'in_invoice'),
-            ('parent_state', '=', 'posted'),
-        ])
-
-    def _get_late_purchases_query(self):
-        return self.env['account.move.line']._where_calc([
-            *self.env['account.move.line']._check_company_domain(self.env.companies),
-            ('journal_id', 'in', self.ids),
-            ('move_id.payment_state', 'in', ('not_paid', 'partial')),
-            ('account_type', '=', 'liability_payable'),
-            ('date_maturity', '<', fields.Date.context_today(self)),
-            ('move_id.move_type', '=', 'in_invoice'),
-            ('parent_state', '=', 'posted'),
+            ('move_id', 'in', self.env['account.move']._where_calc([
+                *self.env['account.move.line']._check_company_domain(self.env.companies),
+                ('journal_id', 'in', self.ids),
+                ('payment_state', 'in', ('not_paid', 'partial')),
+                ('move_type', '=', 'out_invoice' if journal_type == 'sale' else 'in_invoice'),
+                ('state', '=', 'posted'),
+            ])),
+            ('account_type', '=', 'asset_receivable' if journal_type == 'sale' else 'liability_payable'),
         ])
 
     def _count_results_and_sum_amounts(self, results_dict, target_currency):
         """ Loops on a query result to count the total number of invoices and sum
         their amount_total field (expressed in the given target currency).
         amount_total must be signed!
         """
-        # Create a cache with currency rates to avoid unnecessary SQL requests. Do not copy
-        # curr_cache on purpose, so the dictionary is modified and can be re-used for subsequent
-        # calls of the method.
```

Comment:
This diff is just some minor cleanup that I forgot to remove in another commit a while ago, I don't think it's worth making an extra commit for this :shrug: